### ISSUE TITLE
So u just provide the code for stage B?

### ISSUE THREAD

**Issue by abcdvzz** [phecy/SSL-FEW-SHOT#3]

**Comment by phecy** [comment]
for the pre-train stage, we just use standard AMDIM for self supervised  training.[https://github.com/Philip-Bachman/amdim-public](https://github.com/Philip-Bachman/amdim-public)

And the we released some pretrained models.

**Comment by semi-supervised-paper** [comment]
What's the hyper-parameters of AMDIM for Mini-ImageNet? And what's the data augmentation of Mini-ImageNet in AMDIM?

**Comment by phecy** [comment]
@semi-supervised-paper we use  AmdimNet(ndf=192, ndepth=8, nrkhs=1536) to train Mini-ImageNet. For more detail:
```
python train.py \
  --ndf 192 \
  --n_rkhs 1536 \
  --batch_size 480 \
  --tclip 20.0 \
  --n_depth 8 \
```

**Comment by semi-supervised-paper** [comment]
Thanks for your kindly response. I wonder if you have tried other powerful unsupervised learning methods?

**Comment by phecy** [comment]
we do not do experiment with other ssl algorithms, we believe that other ssl algorithms which aim to  learn ask-agnostic representations  can improve the performance of few-shot learning task.

**Comment by abcdvzz** [comment]
Could u pls give us a detailed configuration of the training of the CUB dataset on stage A? For example, input size, epoch, and transformation. I only get an accuracy of 17% following the Imagenet128 configuration.  Thank u.

**Comment by jaemyungkwon** [comment]
Hi, i want to know this code is ssl with meta learning?
and. do you know how to use multi-gpu for your code?

**Comment by DanielaSun-PLUS** [comment]
Hello, with the setting below:

python train_protonet.py --lr 0.0001 --temperature 128   \
--max_epoch 100 --model_type AmdimNet --dataset MiniImageNet \
--init_weights ./saves/initialization/miniimagenet/mini_imagenet_ndf192_rkhs1536_rd8_ssl_cpt.pth  \
--save_path ./MINI_ProtoNet_MINI_1shot_5way/ \
--shot 1  --way 5 --gpu 4 --step_size 10 --gamma 0.5 \
--ndf 192 --rkhs 1536 --nd 8

I got the performance :

Val Best Acc 0.4774, Test Acc 0.4848
Test Acc 0.4848 + 0.0019

so this codes is ssl without meta learning?
